src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -319,12 +227,15 @@ void mp::QemuVirtualMachine::start()
     if (state == State::suspending)
         throw std::runtime_error("cannot start the instance while suspending");
 
+    vm_process = make_qemu_process(
+        desc, ((state == State::suspended) ? mp::make_optional(monitor->retrieve_metadata_for(vm_name)) : mp::nullopt),
+        tap_device_name);
+    set_process_connections();
```

Comment:
I think I'd be a little happier if we had a `set_vm_process()` that would deal with the connections, lower chance of forgetting to `set_process_connections()` after, wdyt? We don't have much of a precedent of this though…

Reply:
Again, I'm not really sure what you mean here.  Are you saying that I should rename the method to `set_vm_process` instead or move all of the creation of the vm_process object and setting up its signal/slot connections all in one method?  Actually, as I type this out, I prefer doing the later and will change that.  But please let me know :grin:

Reply:
@townsend2010 basically, I'm used to setters and getters in Qt code :)

Pseudo-code:
```cpp
void mp::QemuVirtualMachine::start()
{
    // ...
    set_vm_process(std::move(make_qemu_process(...)));
    // ...
}

// ...

void set_vm_process(std::unique_ptr<Process> process)
{
    vm_process = process;
    // set the connections up...
}
```

But I won't block on that ;)

Reply:
Yeah, I'll do something like that.  I think it's cleaner and is done all in one place.

Reply:
Ok, renamed the private method and added the `make_qemu_process` handling in the method to have it all in one place.  I made this a private method instead of an anonymous namespace function because many class members are needed for this.